cfgov/cdntools/backends.py

```diff
@@ -132,19 +115,13 @@ def purge_all(self):
         )
 
 
-@receiver(post_save, sender=Document)
-@receiver(post_save, sender=CFGOVRendition)
-def cloudfront_cache_invalidation(sender, instance, **kwargs):
-    if not settings.ENABLE_CLOUDFRONT_CACHE_PURGE:
-        return
-
-    if not instance.file:
-        return
-
-    url = instance.file.url
+class MockCacheBackend(BaseBackend):
+    def __init__(self, params):
+        super().__init__(params)
+        self.cached_urls = params.get("CACHED_URLS")
```

Comment:
The later is how we were doing it — with a global. I wanted to try to do something to avoid that global. 

Initially I was thinking [we could use the upcoming `DummyBackend`](https://github.com/wagtail/wagtail/blob/df08f999453dc02a674bec93937a38cacc4e4d79/wagtail/contrib/frontend_cache/backends/dummy.py) to replace our own mock cache backend in the future, but it doesn't provide any way to verify purged URLs at all. Because the cache backends are not persistent, `self.urls` will be empty each time. So... I was trying to bridge implementations, and maybe contribute something more like this back to that `DummyBackend` eventually.

We could go back to a global, but I'd kind of prefer the option to provide the storage that tests will check via configuration instead. I don't feel too strongly about continuing to work on this though, if you think it should be a global I'll make it such.

`self.purged_urls` is a good idea — the global used to be `CACHED_URLS` so that's the origin of that variable name.

Reply:
I don't feel super strongly either, so whichever you prefer! FWIW the Wagtail tests verify purged URLs through use of a separate MockBackend and global, [here](https://github.com/wagtail/wagtail/blob/df08f999453dc02a674bec93937a38cacc4e4d79/wagtail/contrib/frontend_cache/tests.py#L420).

Reply:
Switched back to a global in [91398ff](https://github.com/cfpb/consumerfinance.gov/pull/8522/commits/91398ffd3abab454fa1904984af55e992289153d)